src/main/java/network/radicle/jetbrains/radiclejetbrainsplugin/services/RadicleApplicationService.java

```diff
@@ -124,27 +112,41 @@ public ProcessOutput sync(GitRepository root) {
                 Objects.requireNonNull(root.getCurrentBranchName())), root);
     }
 
-    public ProcessOutput executeCommandWithStdin(String workDir, List<String> args, @Nullable GitRepository repo) {
+    public ProcessOutput executeCommandWithStdin(String workDir, String radHome, String radPath, List<String> args,
+                                                 @Nullable GitRepository repo, String stdin) {
         final var settings = settingsHandler.loadSettings();
-        final var radPath = settings.getPath();
-        return executeCommand(radPath, workDir, args, repo, true);
+        final var path = Strings.isNullOrEmpty(radPath) ? settings.getPath() : radPath;
+        final var home = Strings.isNullOrEmpty(radHome) ? settings.getRadHome() : radHome;
+        return executeCommand(path, home, workDir, args, repo, stdin);
     }
 
     public ProcessOutput executeCommand(String workDir, List<String> args, @Nullable GitRepository repo) {
         final var settings = settingsHandler.loadSettings();
         final var radPath = settings.getPath();
-        return executeCommand(radPath, workDir, args, repo, false);
+        final var radHome = settings.getRadHome();
+        return executeCommand(radPath, radHome, workDir, args, repo, "");
+    }
+
+    public ProcessOutput executeCommand(String exePath, String workDir, List<String> args, @Nullable GitRepository repo) {
+        final var settings = settingsHandler.loadSettings();
+        final var radHome = settings.getRadHome();
+        return executeCommand(exePath, radHome, workDir, args, repo, "");
     }
 
     public ProcessOutput executeCommand(
-            String exePath, String workDir, List<String> args, @Nullable GitRepository repo, boolean isDefaultIdentityAction) {
+            String exePath, String radHome, String workDir, List<String> args, @Nullable GitRepository repo, String stdin) {
         ProcessOutput result;
         final var cmdLine = new GeneralCommandLine();
+        var params = "";
+        if (!Strings.isNullOrEmpty(radHome)) {
+            params = "export RAD_HOME=" + radHome + "; " + exePath + " " + String.join(" ", args);
```

Comment:
this is a very implicit way of setting `RAD_HOME` as env var.
Wouldn't it be best to explicitly add it as env var, as is done for `PATH`?